src/dns.rs

```diff
@@ -0,0 +1,76 @@
+use reqwest::{header, Client, Error};
+use serde::{Deserialize, Serialize};
+
+const HETZNER_API_URL: &str = "https://dns.hetzner.com/api/v1";
```

Comment:
I think the DNS provider shouldn't be fixed to Hetzner DNS. Others might be interested in contributing their own DNS providers and we should make that easy to add. Let's call this module `src/hetzner_dns.rs` and during argument parsing we'll figure which DNS module to use depending on what configuration options are set.

Reply:
The app was refactored to support multi dns provider.